### PR TITLE
Fixes placeholder type and adds docs on usage

### PR DESCRIPTION
**Pull request by mwood23**
#### Is this adding or improving a _feature_ or fixing a _bug_?

Improving a feature

#### What's the new behavior?

Corrects the types of the placeholder prop in Slate-React and adds usage docs.

#### How does this change work?

The placeholder takes a React Node and the types originally said string. People have been looking at how to create a custom placeholder on Slack so I PR'ed to fix the types and add to the docs.

_Italicized placeholder_

#### Have you checked that...?

- [x] The new code matches the existing patterns and styles.
- [x] The tests pass with `yarn test`.
- [x] The linter passes with `yarn lint`. (Fix errors with `yarn fix`.)
- [x] The relevant examples still work. (Run examples with `yarn start`.)

#### Does this fix any issues or need any specific reviewers?

Fixes: https://github.com/ianstormtaylor/slate/issues/3781
Reviewers: @timbuckley

### CODE REVIEW

File: site/examples/richtext.js [hunk 1]
```diff
@@ -37,6 +37,7 @@ const RichTextExample = () => {
       <Editable
         renderElement={renderElement}
         renderLeaf={renderLeaf}
+        // See slate-react docs for how to style a custom placeholder.
```
Comment (timbuckley) [inline site/examples/richtext.js:40]: I don't think this comment will compile because it's amongst JSX, I think it's okay to remove.

```suggestion
```

Comment (mwood23) [reply site/examples/richtext.js:40]: These comments are valid if you're commenting within a component block and are invalid if you are commenting in the render block of JSX (need the fancier comments there). I'll go and remove to keep consistency with other examples.

https://codesandbox.io/s/awesome-feistel-1xuiz?file=/src/App.js

**Comment by BrentFarese** [comment]
@mwood23 can you check this against #3725, which adds some types to `slate-react` for the placeholder using a new custom-types paradigm?  Are the types correct given your work?  If not, we can adjust...

**Comment by mwood23** [comment]
@BrentFarese It does relate to it. Based on that PR it looks like `leaf.placeholder` is getting typed as a string (what my PR changes to `ReactNode`). I think this is a decision point about how Slate wants to type the placeholder prop. Technically, it can accept a `ReactNode` (`string | number | ReactElement | ReactFragment | ReactPortal | boolean | null | undefined`). However, maybe you want to limit it to just a `string` because the placeholder element may change a lot over time.

As it stands, there are no other ways to target the placeholder for styling. If there's not going to be many changes to the placeholder element I think it makes sense to make it a `ReactNode`, and if left as a string introduce a way to target it for styling (https://github.com/ianstormtaylor/slate/issues/3780). 

Since I'm not too familiar with future of the project I'm not sure what direction to recommend. I put in the PR because in the code is was being cast as a `ReactNode` and by that I assumed the intent was leave flexibility to the caller.

**Pull request closed without merging**